File: solutions/typescript/resistor-color-trio/1/resistor-color-trio.ts
const colorCode: { [key: string]: number } = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9,
};

export function decodedResistorValue(colors: string[]): string {
    const [color1, color2, color3] = colors;
    const mainValue = colorCode[color1] * 10 + colorCode[color2];
    const resistance = mainValue * Math.pow(10, colorCode[color3]);

    if (resistance >= 1000000000) {
        return `${resistance / 1000000000} gigaohms`;
    }
    if (resistance >= 1000000) {
        return `${resistance / 1000000} megaohms`;
    }
    if (resistance >= 1000) {
        return `${resistance / 1000} kiloohms`;
    }
    return `${resistance} ohms`;
}

File: solutions/lua/resistor-color-trio/1/resistor-color-trio.lua
return {
    decode = function(c1, c2, c3)
        local COLORS = {
            black = 0,
            brown = 1,
            red = 2,
            orange = 3,
            yellow = 4,
            green = 5,
            blue = 6,
            violet = 7,
            grey = 8,
            white = 9,
        }
        local mainValue = COLORS[c1] * 10 + COLORS[c2]
        local resistance = mainValue * 10 ^ COLORS[c3]
        if resistance >= 1000000000 then
            return resistance / 1000000000, 'gigaohms'
        end
        if resistance >= 1000000 then
            return resistance / 1000000, 'megaohms'
        end
        if resistance >= 1000 then
            return resistance / 1000, 'kiloohms'
        end
        return resistance, 'ohms'
    end,
}
